FirebaseStorageSwift/Tests/Integration/StorageAsyncAwait.swift

```diff
@@ -0,0 +1,391 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseAuth
+import FirebaseCore
+import FirebaseStorage
+import FirebaseStorageSwift
+import XCTest
+
+#if swift(>=5.5)
+  @available(iOS 15, *)
+  class StorageAsyncAwait: StorageIntegrationCommon {
+    func testGetMetadata() async throws {
+      let ref = storage.reference().child("ios/public/1mb2")
+      let result = try await ref.getMetadata()
+      XCTAssertNotNil(result)
+    }
+
+    func testUpdateMetadata() async throws {
+      let meta = StorageMetadata()
+      meta.contentType = "lol/custom"
+      meta.customMetadata = ["lol": "custom metadata is neat",
+                             "ちかてつ": "🚇",
+                             "shinkansen": "新幹線"]
+
+      let ref = storage.reference(withPath: "ios/public/1mb2")
+      let metadata = try await ref.updateMetadata(meta)
+      XCTAssertEqual(meta.contentType, metadata.contentType)
+      XCTAssertEqual(meta.customMetadata!["lol"], metadata.customMetadata!["lol"])
+      XCTAssertEqual(meta.customMetadata!["ちかてつ"], metadata.customMetadata!["ちかてつ"])
+      XCTAssertEqual(meta.customMetadata!["shinkansen"],
+                     metadata.customMetadata!["shinkansen"])
+    }
+
+    func testDelete() async throws {
+      let ref = storage.reference(withPath: "ios/public/fileToDelete")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+      _ = try await ref.delete()
```

Comment:
Can we assert the delete was successful?

Reply:
Yes. Added a second delete to confirm and check throw flow.